doc/rst/feat_adat_tx.rst

```diff
@@ -0,0 +1,134 @@
+
+To begin with, ``XUA_AudioHub`` sends two values on the channel - the master clock multiplier and
+the S/MUX setting.
+
+The master clock multiplier is the ratio between the mclk freqency and the sampling frequency. The S/MUX setting is
+1, 2 or 4, depending on the sampling frequency:
+
+.. literalinclude:: ../../lib_xua/src/core/audiohub/xua_audiohub.xc
+   :start-at: /* Calculate what master clock we should be using */
+   :end-before: /* Calculate master clock to bit clock
+
+This is followed by communicating the address of a block of memory holding the audio samples block.
+The ``XUA_AudioHub`` "runs ahead" of the ADAT transmitter task, assembling the next sample block while the
+ADAT transmitter converts the current block into an ADAT stream to transmit over the optical interface.
+
+The ADAT transmitter, once done processing the current block, acknowledges this by sending a data token over the channel
+to ``XUA_AudioHub`` as a handshake mechanism.
+On receiving this handshake, ``XUA_AudioHub`` sends the address of the next block of samples over the channel.
+
+Note that a ``XS1_CT_END`` end token is not sent between blocks of data, leading to the channel remaining open and getting used
+as a streaming channel.
+
+``XUA_AudioHub`` only terminates the connection by sending a ``XS1_CT_END`` token when there's a sampling frequency change
+that requires the ``XUA_AudioHub`` to re-communicate the master clock multiplier and
+the S/MUX setting.
+
+In case of a sampling frequency change, the ``XUA_AudioHub`` receives the pending handshake from the ADAT transmitter,
+followed by sending the ``XS1_CT_END`` token indicating the end of data streaming to the ADAT task.
+A fresh transmission is then started by sending the new master clock multiplier and
+the S/MUX setting to the ADAT transmitter, followed by audio blocks transfer as described above.
+
+:ref:`Communication between AudioHub and ADAT Transmitter<xua_audiohub_adat_tx>` describes the communication between ``XUA_AudioHub`` and the ADAT transmitter:
+
+ .. _xua_audiohub_adat_tx:
+
+ .. figure:: images/xua_audiohub_adat_tx.png
+   :width: 60%
+
+For further details please see the documentation, application notes and examples provided for ``lib_adat``.
+
+Enumerating as a USB Audio device
+---------------------------------
+
+When ADAT TX is enabled, the number of USB OUT channels vary depending on the sampling freq (S/MUX mode). This is
+exposed to the USB host as alternative interfaces, each supporting different channel counts, for the streaming output
+interface.
+The number of alternative interfaces exposed depends on the ``MIN_FREQ`` and ``MAX_FREQ`` supported over the USB interface.
+In the most generic case, where the device supports all sampling rates from 44.1 to 192 kHz, 3 alternative interfaces on the streaming
+output interface are exposed, supporting 16, 12 and 10 output channels respectively.
```

Comment:
this is application dependent. These values are correct for some configs of the 316 board. Is there a way to make these numbers just relevant to ADAT?

Reply:
Instead of mentioning explicit channel counts I've included the snippen from the code where num out channels are calculated. Hope this looks okay.
![Screenshot 2024-10-23 at 13 20 36](https://github.com/user-attachments/assets/eebe7c70-49f1-41e2-bfa1-202df2d68f43)